st2actions/st2actions/query/mistral/v2.py

```diff
@@ -42,23 +43,37 @@ def query(self, execution_id, query_context):
             status, output = self._get_workflow_result(exec_id)
             if output and 'tasks' in output:
                 LOG.warn('Key conflict with tasks in the workflow output.')
+        except requests.exceptions.ConnectionError:
+            msg = 'Unable to connect to mistral.'
+            trace = traceback.format_exc(10)
+            LOG.exception(msg)
+            return (LIVEACTION_STATUS_RUNNING, {'error': msg, 'traceback': trace})
         except:
             LOG.exception('Exception trying to get workflow status and output for '
                           'query context: %s. Will skip query.', query_context)
             raise
 
         result = output or {}
-        result['tasks'] = self._get_workflow_tasks(exec_id)
+        try:
+            result['tasks'] = self._get_workflow_tasks(exec_id)
+        except requests.exceptions.ConnectionError:
+            msg = 'Unable to connect to mistral.'
+            trace = traceback.format_exc(10)
+            LOG.exception(msg)
+            return (LIVEACTION_STATUS_RUNNING, {'error': msg, 'traceback': trace})
+        except:
+            LOG.exception('Unable to get workflow results for '
+                          'query_context: %s. Will skip query.', query_context)
 
         LOG.debug('Mistral query results: %s' % result)
 
         return (status, result)
 
     def _get_execution_tasks_url(self, exec_id):
-        return self._base_url + 'executions/' + exec_id + '/tasks'
```

Comment:
Thanks for the patch.  It's cool.  This makes thing more consistent.  Otherwise, we have URL config that has trailing slash and some don't.  So messy...